src/components/MovieList/components/MovieCard.test.js

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import MovieCard from "./components/MovieCard";
+import {render, waitForElement} from '@testing-library/react';
+
+
+
+describe("MovieCard", () => {
+    let apiFail;
+    let res;
+
+    beforeAll(() => {
+        global.fetch = jest.fn().mockImplementation((input, init) => {
+            return new Promise((resolve) => {
+                if (apiFail) {
+                    resolve({status: 500});
+                } else {
+                    resolve({status: 200, json: () => Promise.resolve(res)});
+                }
+            });
+        });
+    });
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+        MovieCard.mockImplementation((props) => <div>{props?.movie?.title}</div>);
+        apiFail = false;
+        res = [
+            {
+                id: 1,
+                title: "Titanic"
+            }, {
+                id: 2,
+                title: "Batman"
+            }
+        ]
+    });
+
+    it("should match snapshot", async () => {
+        const {container, getByText} = render(
+            <MemoryRouter>
+                <MovieList />
+            </MemoryRouter>
+        );
+
+        await waitForElement(() => getByText("Titanic"));
+
+        expect(container).toMatchSnapshot();
+    });
```

Comment:
testujesz komponent `MovieCard`, więc wyrenderuj tylko `...<MovieCard movie={data} i_pozostałe_atrybuty /> ...` Zmienną data zainicjalizuj tam gdzie `apiFail `i `res `(żeby była dostępna we wszystkich testach), a dane do niej wrzuć w `beforeEach()`. W tym momencie tą robotę robi zmienna `res`. (a `MovieCard `i tak jest mockowane, więc nie jest tym komponentem, który trzeba przetestować)

Samo `MovieCard` nic nie ładuje na początku, więc test snapshota nie będzie asynchroniczny, będzie po prostu `render` i `expect`